### PR TITLE
Cut-off context for excerpt in case it does not have read-more devider. ...

### PR DESCRIPTION
**Pull request by andie-chernysh**
...Refactored code of EffectivePostsHelper a little bit.

### CODE REVIEW

**Comment by matt-riemer** [comment]
This is a good commit, but I'd like you to go back and tweak the logic:

If :length => ... option IS NOT passed:
- if the Read More divider exists, return the content upto the Read more divider
- if there is no Read more divider, output the entire contents of the post, don't hardcode a default length

If :length => ... option IS passed:
- if the Read More divider is present, and less than :length => value, return the content upto the Read more divider
- if there is no Read more divider, output the contents upto :length

File: app/helpers/effective_posts_helper.rb [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
 
   def effective_post_path(post)
     category = post.category.to_s.downcase
-    EffectivePosts.use_category_routes ? "/#{category}/#{post.to_param}" : effective_posts.post_path(post, :category => category.to_s)
+    EffectivePosts.use_category_routes ? "/#{category}/#{post.to_param}" : effective_posts.post_path(post, category: category.to_s)
   end
 
+  def readmore_link(post, options)
+    content_tag(:p, class: 'post-read-more') do
+      link_to((options.delete(:label) || 'Read more...'), effective_post_path(post), options)
```
Comment (matt-riemer) [inline app/helpers/effective_posts_helper.rb]: I think this should be effective_posts.post_path(post)

**Comment by andie-chernysh** [comment]
@matt-riemer I changed logic as you suggested. One thing I am worried of - I was told that read-more divider should override `:length` parameter, but in this case it overrides if it has lesser position only. Is that  ok?

**Comment by matt-riemer** [comment]
Hey Andrew....

I know we've been going back and forth on this.

Every effective_region is going to spit out valid HTML.  If we're grabbing the excerpt text, I don't want to lose the html.  I don't like strip_tags

Could you use this truncate_html() method instead?

https://gist.github.com/matt-riemer/674fe124f2cc42968349

(I wrote this and it was in effective_posts for awhile but removed))

**Comment by andie-chernysh** [comment]
Hi @matt-riemer !
I used that helper you had suggested. Yep, works fine, I guess.

**Pull request merged**